### PR TITLE
Add the scripts to configure Nexus for Cachito JavaScript support

### PR DESCRIPTION
**Pull request by mprahl**
This PR also fixes the Nexus development configuration and a typo in a docstring.

Please see each commit for more details.

### CODE REVIEW

File: cachito/workers/nexus_scripts/js_after_content_staged.groovy [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/*
+This script configures Nexus so that the NPM proxy repository for the Cachito request is blocked from getting additional
+content.
+
+In addition, a temporary user for the request is created and is given permission to access this NPM proxy repository.
+It'd be preferable to give access to the Nexus anonymous user instead, but there is no way to add a role to a user.
```
Comment (lcarva) [inline cachito/workers/nexus_scripts/js_after_content_staged.groovy:6]: Is there a way to use a single role to grant access only to all the request specific repos? Not sure if we can name these repos with a common prefix and use something like `"nx-repository-view-npm-public*-read"` as the role added to the anonymous user.

Comment (mprahl) [reply cachito/workers/nexus_scripts/js_after_content_staged.groovy:6]: I will investigate to see if I can create a [privilege](https://help.sonatype.com/repomanager3/security/privileges#Privileges-Example:CreatingaRepositoryViewPrivilege) that does that.

I've been going with the privileges that Nexus creates for us.

File: docker/configure-nexus.py [hunk 2]
```diff
@@ -32,13 +32,36 @@
...
+    json=realms,
+    timeout=15,
+)
+
+if not rv_set_realms.ok:
+    print(
+        f'Setting the realms to {realms} failed with the status code: {rv_set_realms.status_code}\n'
+        f'The output was: {rv_set_realms.text}',
+        file=sys.stderr,
+    )
```
[the remark this answers is not in the document]
Comment (mprahl) [reply docker/configure-nexus.py:56]: Good catch! I'll fix that.

File: cachito/workers/nexus_scripts/js_after_content_staged.groovy (hunk 1 above)
Comment (mprahl) [reply cachito/workers/nexus_scripts/js_after_content_staged.groovy:6]: So for the privilege `nx-repository-view-npm-cachito-js-1-read`, the permission behind the privilege is `nexus:repository-view:npm:cachito-js-1:read`. It seems like between the colons, you can use the `*` wildcard to denote things such as any repo or any action on the repo, but I haven't figured out how to say read access on all repos with the pattern `cachito-js-\d+`.

There's like no documentation on this, so I'll contact Sonatype support to see if they have any suggestions.

Comment (mprahl) [reply cachito/workers/nexus_scripts/js_after_content_staged.groovy:6]: @lcarva according to Sonatype support, they say this is not possible.

Comment (lcarva) [reply cachito/workers/nexus_scripts/js_after_content_staged.groovy:6]: :sob:

**Pull request merged**